src/MessagingClient.ts

```diff
@@ -76,6 +76,20 @@ export class MessagingClient implements MessagingAPI {
         .map((conv): Conversation => conv.conversation)
     }
 
+    /** Get total number of unseen messages from all conversations the user has joined */
+    getTotalUnseenMessages(): number {
+        const rooms = this.getAllRooms()
+        return rooms
+            .filter(room => room.getMyMembership() === 'join') // Consider rooms that I have joined
+            .map(room => {
```

Comment:
Avoid this `map` altogether and use the `getRoomUnreadMessages` in the `reduce` to iterate through elements once here.